src/ZoneServer/Skills/Handlers/Enchanter/EnchantAura.cs

```diff
@@ -0,0 +1,169 @@
+		{
+			if (!caster.TrySpendSp(skill))
+			{
+				caster.ServerMessage(Localization.Get("Not enough SP."));
+				return;
+			}
+
+			skill.IncreaseOverheat();
+			caster.SetAttackState(true);
+
+			Send.ZC_NORMAL.Skill_88(caster, caster, skill);
+
+
+			var buffComponent = caster.Components.Get<BuffComponent>();
+
+			if (buffComponent != null && buffComponent.Has(BuffId.EnchantAura_Buff))
+			{
+				var x = skill.Vars.GetFloat("Melia.EchantAura.X");
+				var y = skill.Vars.GetFloat("Melia.EchantAura.Y");
+				var z = skill.Vars.GetFloat("Melia.EchantAura.Z");
+				var id = skill.Vars.GetInt("Melia.EchantAura.EffectId");
+
+				this.RemoveSkillEffect(skill, caster, new Position(x, y, z), id);
+				return;
+			}
+
+			var castedPos = caster.Position;
+			var effectId = ForceId.GetNew();
+
+			skill.Vars.SetFloat("Melia.EchantAura.X", castedPos.X);
+			skill.Vars.SetFloat("Melia.EchantAura.Y", castedPos.Y);
+			skill.Vars.SetFloat("Melia.EchantAura.Z", castedPos.Z);
+			skill.Vars.SetInt("Melia.EchantAura.EffectId", effectId);
+
+			Send.ZC_SKILL_READY(caster, skill, caster.Position, caster.Position);			
+			Send.ZC_NORMAL.UpdateSkillEffect(caster, caster.Handle, caster.Position, caster.Position.GetDirection(caster.Position), Position.Zero);
+			Send.ZC_SKILL_MELEE_GROUND(caster, skill, originPos, ForceId.GetNew(), null);
+
+			this.AreaOfEffect(skill, caster, castedPos, effectId);
+		}
+
+		/// <summary>
+		/// Area of effect that ticks dealing damage on the enemies inside
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="position"></param>
+		/// <param name="effectId"></param>
+		private async void AreaOfEffect(Skill skill, ICombatEntity caster, Position position, int effectId)
+		{
+			var buffComponent = caster.Components.Get<BuffComponent>();
+
+			if (buffComponent == null)
+				return;
+
+			await Task.Delay(TimeSpan.FromMilliseconds(200));
+
+			Send.ZC_NORMAL.GroundEffect_59(caster, caster.Direction, "Enchanter_EnchantAura", skill.Id, caster.Position, effectId, true);
+
+			var radius = 80;
+			var center = position.GetRelative(position, radius);
+			var splashArea = new Circle(center, radius);
+
+			Debug.ShowShape(caster.Map, splashArea, edgePoints: false);
+
+			if (buffComponent != null && !buffComponent.Has(BuffId.EnchantAura_Buff))
+			{
+				caster.StartBuff(BuffId.EnchantAura_Buff, TimeSpan.Zero);
+			}
+
+			await Task.Delay(TimeSpan.FromSeconds(1));
+
+			while (buffComponent.Has(BuffId.EnchantAura_Buff))
+			{
+				await Task.Delay(2000);
+
+				// Cancel if the caster has not enough SP
+				if (!caster.TrySpendSp(54f))
+				{
+					this.RemoveSkillEffect(skill, caster, position, effectId);
+					break;
+				}
+
+				var character = caster as Character;
+
+				if ((character != null && !character.Connection.LoggedIn) || caster.IsDead)
+				{
+					break;
```

Comment:
It also needs to remove the skill effect here.  You might want to move the RemoveSkillEffect from the previous bracket to the end of the loop, as in all of the situations where the execution escapes the loop (these two brackets and the while condition failing) the effect should be removed.